storage3/types.py

```diff
@@ -41,9 +41,11 @@ class ListBucketFilesOptions(TypedDict):
 
 
 class TransformOptions(TypedDict):
-    height: Optional[float]
-    width: Optional[float]
+    height: Optional[int]
+    width: Optional[int]
     resize: Optional[Union[Literal["cover"], Literal["contain"], Literal["fill"]]]
+    format: Optional[Union[Literal["origin"], Literal["avif"]]]
+    quality: Optional[int]
```

Comment:
Are all fields _required_ here? The `Optional` type hint doesn't really make a key here optional (very bad naming 😓 ), but only says that it can be `type | None` - i.e A user would be forced to pass something like
```py
{"height": None, "width": None, "resize": None, "format": None, "quality": 10}
``` 
if they wanted to only use one transform instead of just passing `{"quality": 10}`
If non-required fields can be skipped, pass `total=False` to the typeddict ([doc](https://docs.python.org/3/library/typing.html#typing.TypedDict)
```py
class TransformOptions(TypedDict, total=False):
    height: int
    width: int
    ... # and so on - the types will no longer be Optional
```

Reply:
The same comment applies to the CreateSignedURLOptions type too - if all fields aren't required, please pass `total=False` there too and remove the Optional from every field

Reply:
Fixed this and added a separate `CreateSignedURLsOptions` class since Supabase's API currently doesn't support transforms when retrieving more than one signed URL.